src/shelly_temp_sensor_ow.cpp

```diff
@@ -0,0 +1,212 @@
+    sensors->push_back(std::move(sensor));
+    num_sensors++;
+  }
+  LOG(LL_INFO, ("Discovered %i sensors", num_sensors));
+}
+
+std::unique_ptr<OWTempSensor> Onewire::NextAvailableSensor(int type) {
+  uint8_t rom[8] = {0};
+  int mode = 0;
+  std::unique_ptr<OWTempSensor> sensor;
+  if (mgos_onewire_next(ow_, rom, mode)) {
+    uint8_t family = rom[0];
+    if (TempSensorDS18XXX::SupportsFamily(family)) {
+      sensor.reset(new TempSensorDS18XXX(ow_, rom));
+    } else {
+      LOG(LL_INFO, ("Found non-supported device"));
+    }
+  }
+  (void) type;
+  return sensor;
+}
+
+OWTempSensor::OWTempSensor(struct mgos_onewire *ow, uint8_t *rom)
+    : cached_temperature_(0),
+      meas_timer_(std::bind(&OWTempSensor::UpdateTemperatureCB, this)),
+      read_timer_(std::bind(&OWTempSensor::ReadTemperatureCB, this)) {
+  memcpy(rom_, rom, 8);
+  ow_ = ow;
+}
+
+OWTempSensor::~OWTempSensor() {
+}
+
+void OWTempSensor::StartUpdating(int interval) {
+  meas_timer_.Reset(interval, MGOS_TIMER_REPEAT | MGOS_TIMER_RUN_NOW);
+}
+
+TempSensorDS18XXX::TempSensorDS18XXX(struct mgos_onewire *ow, uint8_t *rom)
+    : OWTempSensor(ow, rom) {
+  resolution_ = GetResolution();
+  uint64_t serial_mask = 0xFFFFFFFFFFFF;
+  uint64_t serial_ = ((*(uint64_t *) rom) >> 8) & serial_mask;
+  LOG(LL_INFO, ("DS18XXX: model: %02X, serial number: %" PRIx64
+                ", resolution: %d, parasitic power: %d",
+                rom[0], serial_, resolution_, ReadPowerSupply()));
+}
+
+TempSensorDS18XXX ::~TempSensorDS18XXX() {
+}
+
+StatusOr<float> TempSensorDS18XXX::GetTemperature() {
+  return cached_temperature_;
+}
+
+bool TempSensorDS18XXX::SupportsFamily(uint8_t family) {
+  if (family == DS18B20MODEL or family == DS18S20MODEL or
+      family == DS1822MODEL or family == DS1825MODEL) {
+    return true;
+  }
+  return false;
+}
+
+void TempSensorDS18XXX::ReadScratchpad(uint8_t *scratchpad, size_t len) {
+  if (!mgos_onewire_reset(ow_)) {
+    return;
+  }
+  mgos_onewire_select(ow_, rom_);
+  mgos_onewire_write(ow_, READ_SCRATCHPAD);
+  mgos_onewire_read_bytes(ow_, scratchpad, len);
+}
+
+bool TempSensorDS18XXX::ReadPowerSupply() {
+  if (!mgos_onewire_reset(ow_)) {
+    return false;
+  }
+  mgos_onewire_select(ow_, rom_);
+  mgos_onewire_write(ow_, READ_POWER_SUPPLY);
+  return (mgos_onewire_read_bit(ow_) == 0);
+}
+
+bool TempSensorDS18XXX::VerifyScratchpad(uint8_t *scratchpad) {
+  return mgos_onewire_crc8(scratchpad, 8) == scratchpad[SCRATCHPAD_CRC];
+}
+
+void TempSensorDS18XXX::ReadTemperatureCB() {
+  uint8_t temp[SCRATCHPAD_SIZE];
+  ReadScratchpad(temp, SCRATCHPAD_SIZE);
+  if (VerifyScratchpad(temp)) {
+    uint16_t temp_s = temp[1] << 8 | temp[0];
+    float temperature = temp_s * 0.0625;
+    LOG(LL_INFO, ("Read temperature %f", temperature));
+    cached_temperature_ = temperature;
+    if (notifier_) {
+      notifier_();
+    }
+  }
+}
+
+void TempSensorDS18XXX::UpdateTemperatureCB() {
+  LOG(LL_INFO, ("Update temperature"));
```

Comment:
please reduce INFO level verbosity. as a general rule, only meaningful state changes (like user actions) should log at INFO level. if nothing is happening, only the periodic status timer should be logging messages (every 8 seconds). it's ok to add sensor readings to that but noise at info level should be avoided.